src/sonic-config-engine/minigraph.py

```diff
@@ -484,6 +488,39 @@ def parse_xml(filename, platform=None, port_config_file=None):
 
     return results
 
+def parse_device_desc_xml(filename):
+    root = ET.parse(filename).getroot()
+    (lo_prefix, mgmt_prefix, hostname, hwsku, d_type) = parse_device(root)
+
+    Tree = lambda: defaultdict(Tree)
+    results = Tree()
+    results['minigraph_hwsku'] = hwsku
+    results['minigraph_hostname'] = hostname
+    results['inventory_hostname'] = hostname
+
+    lo_intfs = []
+    ipn = ipaddress.IPNetwork(lo_prefix)
+    ipaddr = ipn.ip
+    prefix_len = ipn.prefixlen
+    ipmask = ipn.netmask
+    lo_intf = {'name': None, 'addr': ipaddr, 'prefixlen': prefix_len}
+    if isinstance(ipn, ipaddress.IPv4Network):
+        lo_intf['mask'] = ipmask
+    else:
+        lo_intf['mask'] = str(prefix_len)
+    lo_intfs.append(lo_intf)
+    results['minigraph_lo_interfaces'] = lo_intfs
+
+    mgmt_intf = None
+    mgmtipn = ipaddress.IPNetwork(mgmt_prefix)
```

Comment:
Fixed.

---
In reply to: [125150747](https://github.com/Azure/sonic-buildimage/pull/775#discussion_r125150747) [](ancestors = 125150747)